app/controlplane/pkg/ca/ejbca/ejbca.go

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ejbca
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/ca"
+	fulcioca "github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/identity"
+)
+
+type EJBCA struct {
+	// Connection settings
+	// EJBCA installation. ie https://localhost/ejbca/
+	serverURL string
+	// client private key used for authentication
+	keyPath string
+	// client certificate used for authentication
+	certPath string
+	// root CA
+	rootCAPath string
+
+	// Entity Configuration settings
+	// https://docs.keyfactor.com/signserver/latest/tutorial-signserver-container-signing-with-cosign#id-(6.3latest)Tutorial-SignServerContainerSigningwithCosign-Step2-Issuesigningcertificate
+	certificateProfileName string
+	endEntityProfileName   string
+	caName                 string
+}
+
+var _ ca.CertificateAuthority = (*EJBCA)(nil)
+
+func New(serverURL, keyPath, certPath, rootCAPath, certProfileName, endEntityProfileName, caName string) *EJBCA {
```

Comment:
as mentioned somewhere else, it might make sense for this signature to return error

It will allow us to validate that the required options are provided.